pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -48,6 +48,14 @@ type lastRealized struct {
 	ofID uint32
 	// The desired state of a policy rule.
 	*CompletedRule
+	// The OFPort set we have realized for target Pods. We need to record them
+	// because this info will be removed from InterfaceStore after CNI DEL, we
+	// can't know which OFPort to delete when deleting a Pod from the rule. So
+	// we compare the last realized OFPorts and the new desired one to identify
+	// difference.
+	podOFPorts sets.Int32
```

Comment:
This is not relevant to this PR, and is to fix another bug?

Reply:
It fixes another bug and is also relevant to this PR: 
Before the PR, we wait for Pod gets IP so the OFPort and IP address in InterfaceCache must match the one in kube-apiserver, but now in case that containers fail because of non CNI issues and kubelet might call CNI ADD more than once, we need to remove the previous OFPort/IP and add the new one, so need to cache the last realized OFPorts and IPAddress too.

Reply:
Ok. Do you like add this case to comments?

Reply:
Done